### ISSUE TITLE
Huge delay and inaccurate

### ISSUE THREAD

**Issue by JasonKwokGJQ** [ESPresense/ESPresense#454]
I have set up 5 pcs M5Stamp Pico as stations and tested with two beacons(one Android phone and one miband3). Unfortunately, I found it literally unusable due to a huge delay in updating the distance and bad accuracy.

The miband3 needs 10s to update the distance, the interval it shows on MQTT explorer is around 2000-3000, maybe that's the problem but I have no idea how to reduce it.

The interval of my phone is just 200-300 but it also has a 5s delay and the accuracy is quite bad, when I put my phone close to the station it also shows a 1m distance

I wonder if I miss something or I have some wrong configuration but I have no clue, what the performance of this project should be?

**Comment by DTTerastar** [comment]
Yeah an internal of every 3 seconds seeing an advertisement is super slow.  RSSI jumps around a lot so we need multiple readings to get an accurate value.

What kind of phone are we talking?  If using an iBeacon you need adjust your calibrated rssi at 1 m in order to get the correct 1m distance.

What this project does is just some math based on the rssi received.  It does put the value through 2 filters.  A 3 value median filter (the middle value of the last 3 values is used) plus a 1 Euro filter that reduces jitter if not moving but does allow large values (movement) to take effect quickly.  It's open source so you can try messing w/ the filters for your application.  It is not quick enough generally to turn on lights, people typically use a motion sensor for that.  It's used to keep the light on while you are in the room since motion sensors do a poor job of that.  People have used the distance to "follow" them for alexa/google announcements which a bit of latency can still work.
